src/main/java/bio/terra/service/filedata/google/firestore/FireStoreDirectoryDao.java

```diff
@@ -104,8 +102,7 @@ public void createDirectoryEntry(
         String lookupDirPath = makeLookupPath(createEntry.getPath());
 
         fireStoreUtils.runTransactionWithRetry(firestore,
-            (xn) -> {
-
+            (Transaction.Function<Void>) xn -> {
```

Comment:
I was looking at this more and it turns out you can omit the `(Transaction.Function<Void>)` cast, if you don't mind the inferred return type being `Object` instead of `Void`. Java infers the type from the type of the `return` expression.